### PR TITLE
OMWAPPI-1013: LGI reimplementation of RDKNetwork plugin

### PR DESCRIPTION
**Pull request by piotr-dalek-red**
Talks to LGI's Network Configuration Service via dbus and services connection and configuration notifications and some of configuration data.

Signed-off-by: Piotr Dalek <piotr.dalek@consult.red>

### CODE REVIEW

File: LgiNetwork/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# limitations under the License.
+
+set(PLUGIN_NAME Network)
+set(MODULE_NAME ${NAMESPACE}${PLUGIN_NAME})
+
+find_package(${NAMESPACE}Plugins REQUIRED)
+find_package(${NAMESPACE}Definitions REQUIRED)
+find_package(LibGlib REQUIRED)
+find_package(LibGio REQUIRED)
+find_package(DS)
```
Comment (mikolaj-staworzynski-red) [inline LgiNetwork/CMakeLists.txt:25]: Is it needed, Please check: https://github.com/LibertyGlobal/rdkservices/pull/200/commits

File: LgiNetwork/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+add_library(${MODULE_NAME} SHARED
+        Network.cpp
+        NetUtils.cpp
+        NetUtilsNetlink.cpp
+        NetworkTraceroute.cpp
+        PingNotifier.cpp
+        Module.cpp
+        dbus/lginetwork_client.cpp
+        dbus/lginetwork_dbus_api.c
+        utils.cpp
```
Comment (mikolaj-staworzynski-red) [inline LgiNetwork/CMakeLists.txt:43]: Is this module really needed (utils.cpp), please check: https://github.com/LibertyGlobal/rdkservices/pull/200/commits

File: LgiNetwork/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+add_definitions (-DNET_DEFINED_INTERFACES_ONLY)
+
+include_directories(${LIBGLIB_INCLUDE_DIRS})
+include_directories(${LIBGIO_INCLUDE_DIRS})
+
+target_link_libraries(${MODULE_NAME}
+        PRIVATE
+                ${NAMESPACE}Plugins::${NAMESPACE}Plugins
+                ${NAMESPACE}Definitions::${NAMESPACE}Definitions
+                ${IARMBUS_LIBRARIES}
```
Comment (mikolaj-staworzynski-red) [inline LgiNetwork/CMakeLists.txt:59]: Is it needed ? IARMBUS_LIBRARIES

File: LgiNetwork/Network.cpp [hunk 4]
```diff
@@ -0,0 +1,995 @@
...
+                m_NetworkClient.onStatusChangeEvent = StatusChangeEvent;
+                m_NetworkClient.onNetworkingEvent = NetworkingEvent;
+                LOGINFO("Succeeded initializing LGI DBus Network Client");
+            }
+            return msg;
+        }
+
+        void Network::Deinitialize(PluginHost::IShell* /* service */)
+        {
+            m_isPluginInited = false;
```
Comment (mikolaj-staworzynski-red) [inline LgiNetwork/Network.cpp:118]: IMHO we need some deinitialization for: m_NetworkClient.Run(); ?

see the following in logs while plugin Initialization 2nd time:
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: [8] INFO [lginetwork_client.cpp:288] Run: user_data: 0x133e1fc
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-GObject-WARNING **: 12:33:42.781: cannot register existing type 'Networkconfig1Proxy'
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-GObject-WARNING **: 12:33:42.782: cannot add private field to invalid (non-instantiatable) type '<invalid>'
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-GObject-WARNING **: 12:33:42.782: cannot register existing type 'Networkconfig1'
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-GObject-CRITICAL **: 12:33:42.782: g_type_interface_add_prerequisite: assertion 'G_TYPE_IS_INTERFACE (interface_type)' failed
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-CRITICAL **: 12:33:42.782: g_once_init_leave: assertion 'result != 0' failed
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-GObject-CRITICAL **: 12:33:42.782: g_type_add_interface_static: assertion 'G_TYPE_IS_INSTANTIATABLE (instance_type)' failed
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-CRITICAL **: 12:33:42.782: g_once_init_leave: assertion 'result != 0' failed
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-GIO-CRITICAL **: 12:33:42.782: g_initable_new_valist: assertion 'G_TYPE_IS_INITABLE (object_type)' failed

**Review by mikolaj-staworzynski-red — CHANGES_REQUESTED** [review]

File: LgiNetwork/Network.cpp [hunk 5]
```diff
@@ -0,0 +1,1086 @@
...
+                    result = true;
+                }
+                else
+                {
+                    LOGWARN ("Network plugin not initialised yet returning from %s", __FUNCTION__);
+                }
+
+                returnResponse(result)
+            }
+            return 0;
```
Comment (mikolaj-staworzynski-red) [inline LgiNetwork/Network.cpp:305]: It should return WPEFramework::Core::ERROR_*, probably WPEFramework::Core::ERROR_GENERAL here

Comment (piotr-dalek-red) [reply LgiNetwork/Network.cpp]: Changed

**Review by mikolaj-staworzynski-red — APPROVED** [review]

**Pull request merged**